tests/test_y_text.py

```diff
@@ -110,3 +110,58 @@ def callback(e):
         x.insert(txn, 1, "fgh")
     assert target == None
     assert delta == None
+
+
+def test_delta_embed_attributes():
+
+    d1 = Y.YDoc()
+    text = d1.get_text("test")
+
+    delta = None
+
+    def callback(e):
+        nonlocal delta
+        delta = e.delta
+
+    sub = text.observe(callback)
+
+    with d1.begin_transaction() as txn:
+        text.insert(txn, 0, "ab", {"bold": True})
+        text.insert_embed(txn, 1, {"image": "imageSrc.png"}, {"width": 100})
+
+    expected = [
+        {"insert": "a", "attributes": {"bold": True}},
+        {"insert": {"image": "imageSrc.png"}, "attributes": {"width": 100}},
+        {"insert": "b", "attributes": {"bold": True}},
+    ]
+    assert delta == expected
+
+    text.unobserve(sub)
+
+
+def test_formatting():
+    d1 = Y.YDoc()
+    text = d1.get_text("test")
+
+    delta = None
+    target = None
+
+    def callback(e):
+        nonlocal delta
+        nonlocal target
+        delta = e.delta
+        target = e.target
+
+    sub = text.observe(callback)
+
+    with d1.begin_transaction() as txn:
+        text.insert(txn, 0, "stylish")
+        text.format(txn, 0, 4, {"bold": True})
+
+    expected = [
+        {"insert": "styl", "attributes": {"bold": True}},
+        {"insert": "ish"},
+    ]
+    assert delta == expected
```

Comment:
@Horusiath I expected this sort of structure when formatting the `YText` in a single transaction, but I'm getting `{"insert": "stylish"}` instead. Is this expected behavior?

Reply:
I don't think so.